processing/src/main/java/org/apache/druid/frame/processor/OutputChannelFactory.java

```diff
@@ -31,6 +31,11 @@
    */
   OutputChannel openChannel(int partitionNumber) throws IOException;
 
+  /**
+   * Create a channel pair tagged with a particular name and a flag to delete the channel data after its read.
+   */
+  PartitionedOutputChannel openPartitionedChannel(String name, boolean deleteAfterRead) throws IOException;
```

Comment:
I have a similar comment like @gianm . I think the two interface calls looks very similar.  
Couple of questions:
1.  Does partitionName have to be a string and not an int or another way is why not work with partition numbers only ?
2. From the javadoc, the pair seems to be of a OutputChannel and the partition name and if the channel data needs to be deleted on exit.  My question is the called would be interested in the output channel for that "partition" and clean up. So why do we need another API for this. We could have a cleaning output channel factory impl which could do the cleaning for us no ?

Reply:
1. the name is not a partition name, but rather a generic name for the channel which can contain multiple partitions.
2. FrameFiles can be shared amongst multiple independent readers currently - and they are deleted only when all references to the file are removed. So, that's why had to pass the deletion flag down to the channel creator. A cleaning factory could be created but then it would also have to do ref counting per frame file before cleaning.